pkg/virt-handler/device-manager/usb/manager.go

```diff
@@ -0,0 +1,449 @@
+		queue:              queue,
+		discoveryFunc:      discoverUSBDevices,
+		factoryFunc:        NewUSBDevicePlugin,
+		state:              newState(),
+		logger:             log.Log.With("subcomponent", "usb-manager"),
+	}
+	nodeConfigInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc:    manager.addFunc,
+		UpdateFunc: manager.updateFunc,
+		DeleteFunc: manager.deleteFunc,
+	})
+
+	return manager
+}
+
+func (manager *USBManager) cleanUpWorker(stop chan struct{}) func() {
+	return func() {
+		t := time.NewTicker(5 * time.Second)
+		defer t.Stop()
+		for {
+			select {
+			case <-t.C:
+				for _, nodeConfig := range manager.state.list() {
+					if _, exists, _ := manager.nodeConfigInformer.GetIndexer().GetByKey(nodeConfig); !exists {
+						manager.logger.Infof("Cleaning up plugin for %s node config", nodeConfig)
+						manager.state.clean(nodeConfig)
+					}
+				}
+			case <-stop:
+				return
+			}
+		}
+	}
+}
+
+func (manager *USBManager) Run(stopCh chan struct{}) {
+	defer manager.queue.ShutDown()
+
+	manager.logger.Info("Starting USB manager")
+
+	cache.WaitForCacheSync(stopCh, manager.nodeConfigInformer.HasSynced)
+
+	// Start the actual work
+	go wait.Until(manager.runWorker, time.Second, stopCh)
+	// TODO: This might not be necessary as execute() should know when a nodeConfig changed or was
+	// deleted.
+	go wait.Until(manager.cleanUpWorker(stopCh), time.Second, stopCh)
+	manager.logger.Info("Started USB manager")
+
+	<-stopCh
+	manager.logger.Info("Stoping USB manager")
+}
+
+func (manager *USBManager) Execute() bool {
+	key, quit := manager.queue.Get()
+	if quit {
+		return false
+	}
+	defer manager.queue.Done(key)
+
+	if err := manager.execute(key.(string)); err != nil {
+		manager.logger.Reason(err).Infof("re-enqueuing NodeConfig %v", key)
+		manager.queue.AddRateLimited(key)
+	} else {
+		manager.logger.V(5).Infof("processed NodeConfig %v", key)
+		manager.queue.Forget(key)
+	}
+	return true
+}
+
+func (manager *USBManager) runWorker() {
+	for manager.Execute() {
+	}
+}
+
+func (manager *USBManager) execute(key string) error {
+	obj, exists, err := manager.nodeConfigInformer.GetStore().GetByKey(key)
+	if err != nil {
+		return fmt.Errorf("failed to get object for key %s, %v", key, err)
+	}
+
+	if !exists || obj == nil {
+		manager.state.clean(key)
+		return nil
+	}
+
+	// If key already exists, cleanup before proceeding
+	manager.state.clean(key)
+
+	manager.logger.V(5).Infof("Iterating over %s", key)
+	nodeConfig := obj.(*v1alpha1.NodeConfig)
+	return manager.syncDevicePlugin(nodeConfig, key)
+}
+
+func constructPermittedUSBDevicesMap(nodeConfig *v1alpha1.NodeConfig) map[int][]usbDeviceSelector {
+	// Iterate over requested USB Devices and map it vendor:product
+	permittedUSBDevices := make(map[int][]usbDeviceSelector)
+	resourceName := nodeConfig.Spec.USB.ResourceName
+	for index, dev := range nodeConfig.Spec.USB.USBHostDevices {
+		sep := strings.Index(dev.SelectByVendorProduct, ":")
```

Comment:
If you use [strings.Split](https://pkg.go.dev/strings#Split), you can directly split the string in 2 parts, verify that `length == 2` and then refer for the vendor as index 0 and product index 1. Not sure, but it seems a simpler to me :). You can have a try what you like most